nntrainer/layers/multi_head_attention_layer.cpp

```diff
@@ -55,28 +54,257 @@ enum AttentionParams {
   fc_weight,
   fc_bias,
   projected_query,
-  d_projected_query,
   projected_key,
-  d_projected_key,
   projected_value,
-  d_projected_value,
-  transposed_projected_query,
-  transposed_projected_key,
-  transposed_projected_value,
   attention_score,
   d_attention_score,
-  attention_mask,
+  // attention_mask,
   attention_weight,
   dropout_mask,
   attention_output,
-  d_attention_output,
-  transposed_attention_output,
 };
 
-void MultiHeadAttentionLayer::finalizeCommon(InitLayerContext &context) {}
-
 void MultiHeadAttentionLayer::finalize(InitLayerContext &context) {
-  finalizeCommon(context);
+  if (context.getNumInputs() < 3 || context.getNumInputs() > 4) {
+    throw std::runtime_error("Multi head Attention layer needs 3 or 4 inputs. "
```

Comment:
invalid argument with NNTR_THROW_IF ?

Reply:
Done

Reply:
I will make a another pr for other layers